plugin/scripts/pre-tool-use.sh

```
@@ -128,17 +114,27 @@ fi
 echo "$CHANGE_CONTEXT" > "$PENDING_CHANGE_PATH"
 
-# Run evaluation with transcript context
-if [ -n "$SESSION_ID" ]; then
-    log "Running: sg evaluate-llm --session-id $SESSION_ID with pending change"
-    RESULT=$(sg evaluate-llm --transcript-path "$TRANSCRIPT_PATH" --session-id "$SESSION_ID" 2>> .superego/hook.log)
+# Run evaluation with transcript context (redirect all output to log)
+# Atomic lock to prevent duplicate evaluations
+LOCK_FILE="$SESSION_DIR/eval.lock"
+if mkdir "$LOCK_FILE" 2>/dev/null; then
+    # Got lock, run evaluation
+    trap "rmdir '$LOCK_FILE' 2>/dev/null" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)